### PR TITLE
Make notification file test more robust

### PR DESCRIPTION
**Pull request by tej99**
# Motivation and Context

Currently only on enrolment code was getting checked, now we check
all of the expected enrolment code. 
Also the enrolment codes sometimes get written over multiple notification files, now we make sure we check all files instead of just one.

# What has changed

Now checking all enrolment codes are in the notification file.
Also check all notification files for that survey ref and 
collection exercise instead of just one.

# How to test?

Ran ATs

# Links

### CODE REVIEW

File: acceptance_tests/features/steps/notification_letter.py [hunk 1]
```diff
@@ -1,16 +1,18 @@
 import logging
 import time
+from copy import copy
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/notification_letter.py]: unused import

File: acceptance_tests/features/steps/notification_letter.py [hunk 2]
```diff
@@ -64,24 +66,35 @@ def _get_sftp_client():
...
+                if iac in content:
+                    seen_iacs.append(iac)
+
+    if set(seen_iacs) != set(expected_iacs):
+        file_names = [f.filename for f in files]
+        logger.info('Unable to find all iacs', files_found=file_names, seen_iacs=seen_iacs, expected_iacs=expected_iacs)
         raise FileNotFoundError
 
-    return f'{Config.SFTP_DIR}/{filename}'
+    for file in files:
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/notification_letter.py]: This is potentially problematic. We don't know _which_ files the IAC codes were in, and we could end up deleting files which didn't have *any* IAC codes we expected in them, which might hamper test debugging

File: controllers/database_controller.py [hunk 3]
```diff
@@ -61,3 +62,40 @@ def get_iac_for_collection_exercise(collection_exercise_id, social=False):
     for row in result:
         iac = row['iac']
     return iac
+
+
+def get_all_iacs_for_collection_exercise(collection_exercise_id, social=False):
```
Comment (NickGrantONS) [inline controllers/database_controller.py]: is `get_iac_for_collection_exercise` now dead unused code?

File: acceptance_tests/features/steps/notification_letter.py (hunk 2 above)
Comment (tej99) [reply acceptance_tests/features/steps/notification_letter.py]: Okay, is removing the delete the files here and 'reseting' the sftp server at the start of the ATs better?

File: controllers/database_controller.py (hunk 3 above)
Comment (tej99) [reply controllers/database_controller.py]: Nope used in multiple places

**Pull request merged**